pandas/core/algorithms.py

```diff
@@ -944,14 +944,27 @@ def select_n_frame(frame, columns, n, method, keep):
     -------
     nordered : DataFrame
     """
-    from pandas.core.series import Series
     if not is_list_like(columns):
         columns = [columns]
-    columns = list(columns)
-    ser = getattr(frame[columns[0]], method)(n, keep=keep)
-    if isinstance(ser, Series):
-        ser = ser.to_frame()
-    return ser.merge(frame, on=columns[0], left_index=True)[frame.columns]
+    else:
+        columns = list(columns)
+    reverse = method == 'nlargest'
+    for i, column in enumerate(columns):
+        series = frame[column]
+        if reverse:
```

Comment:
But if there are duplicates I need to do another filter using ``isin`` this returns an **unsorted** view of the series, I then have no way of aligning the series according to the sorting requirement, i'd still need to do an argsort after the ``nsmallest``/``nlargest`` call